en/concept/architecture.md

```diff
@@ -1,36 +1,121 @@
 # Architectural Overview
 
-PX4 consists of two main layers: The [PX4 flight stack](../concept/flight_stack.md), an autopilot software solution and the [PX4 middleware](../concept/middleware.md), a general robotics middleware which can support any type of autonomous robot.
+PX4 consists of two main layers: the [flight stack](#flight-stack) is an autopilot software solution,
```

Comment:
It might be nice, if you can find the words, to describe in just a little more "specificity" what the flight stack offers. Saying it is an autopilot is a bit pointless, since that definition can also apply to the whole of PX4. Perhaps something like "flight control and optimisation system".